src/Nethermind/Nethermind.Facade/BlockchainBridge.cs

```diff
@@ -191,7 +194,39 @@ public CallOutput EstimateGas(BlockHeader header, Transaction tx, CancellationTo
                 estimateGasTracer.WithCancellation(cancellationToken));
             
             long estimate = estimateGasTracer.CalculateEstimate(tx);
-            return new CallOutput {Error = tryCallResult.Success ? estimateGasTracer.Error : tryCallResult.Error, GasSpent = estimate, InputError = !tryCallResult.Success};
+            
+            return new CallOutput 
+            {
+                Error = tryCallResult.Success ? estimateGasTracer.Error : tryCallResult.Error, 
+                GasSpent = estimate, 
+                InputError = !tryCallResult.Success
+            };
+        }
+
+        public CallOutput CreateAccessList(BlockHeader header, Transaction tx, CancellationToken cancellationToken, bool optimize)
+        {
+            CallOutputTracer callOutputTracer = new();
+            AccessTxTracer accessTxTracer = optimize 
```

Comment:
always optimize, do you want to tell me that the API from Geth has an unoptimized version?

Reply:
why do we add GetRecipient here? logic seems to be split into too many places

Reply:
optimize - is for returning AccessList with lowest cost if some storage is involved on Sender or Recipient addresses. Will add some comment to EIP